File: daily/30 -day finish 300 coding challenge/2020-5-13. 10 questions - Two Pointer/26. Remove Duplicates from Sorted Array.cpp
#include<vector>
using namespace std;
class Solution{
public:
    int removeDuplicates(vector<int>& nums){
        if (nums.size() == 0) return 0;
        int i = 0;
        for (int c = 1; c < nums.size(); c++){
            if (nums[i] != nums[c]){
                i ++;
                nums[i] = nums[c];
            }
        }
        return i + 1;
    }
};

File: daily/30 -day finish 300 coding challenge/2020-5-13. 10 questions - Two Pointer/26. Remove Duplicates from Sorted Array.py
class Solution:
    def removeDuplicates(self, nums):
        if not nums: return 0
        i = 0
        if j in range(1, len(nums)):
            if nums[i] != nums[j]:
                i += 1
                nums[i] = nums[j]
        return i + 1

# time complexity o(n)
# space complexity o(1)
